### ISSUE TITLE
Preflight request is not passing Authorization header

### ISSUE THREAD

**Issue by macmichael01** [axios/axios#5240]
### Describe the bug

My preflight request appears to require an Authorization header but it's not getting set when the preflight request is made. I see a 401 response status for the preflight. 

Preflight:

(Reason: CORS preflight response did not succeed). Status code: 401.
CORS preflight did not succeed

and for the actual request I see:

(Reason: CORS request did not succeed). Status code: (null).
NS_ERROR_DOM_BAD_URI

### To Reproduce

Refer to code snippet.

### Code snippet

Here is the code I am using to construct my request:

~~~javascript
const api = axios.create({
  baseURL: process.env.REACT_APP_API_URL,
  headers: {
    "Content-Type": "application/json",
  },
});

api.interceptors.request.use(
  (config) => {
    const newConfig = { ...config };
    if (AuthService.getAccessToken()) {
      newConfig.headers.common.Authorization = `JWT ${AuthService.getAccessToken()}`;
    }
    return newConfig;
  },
  (error) => Promise.reject(error);
);
~~~

~~~javascript
// Calling this triggers a preflight request but not authorization header
// is getting set despite that I set and defined it above.
await api.get("/some/endpoint");
~~~

### Expected behavior

Preflight should have an Authorization header attached when making the request.

### Axios Version

1.1.3

### Browser

FireFox

### Browser Version

106.0.5

### Node.js Version

14.20

### Additional Library Versions

```bash
React: 18.2.0
```

### Additional context/Screenshots

![Screen Shot 2022-11-09 at 2 17 31 PM](https://user-images.githubusercontent.com/128222/200933172-4f173a0d-ab20-4e63-ac38-d76e6286f91f.png)

**Comment by macmichael01** [comment]
I solved my issue!

**Comment by jasonsaayman** [comment]
@macmichael01 please can you share so other users that have the same issue can use your solution?

**Comment by macmichael01** [comment]
The issue I ran into was with authenticated endpoints. On the server side, `Access-Control-Allow-Origin` header cannot be set to `Access-Control-Allow-Origin = "*"` It must also  be explicitly set to the domain making the request. I also learned that `Access-Control-Allow-Origin` cannot be a list of values either. Use one and only one domain name.

If you are using a non-authenticated endpoint, `Access-Control-Allow-Origin = "*"` is fine (though not recommended), but for authenticated endpoints star is not fine.

So if you see preflight related error from an authenticated endpoint it might be related to the issue I ran into.
